chapter2.adoc

```diff
@@ -1,214 +1,207 @@
-
-[NOTE]
-As the `prv` and `v` fields in dcsr are Write Any Read Legal (WARL) fields, the debugger has two options to confirm the success of a prior write: either by reading back the attempted written value or by checking the `cmderr`, depending on the hardware implementation choice.
-
-If register access without halting the hart is supported, this access bears the privilege of supervisor/hypervisor mode to access the hart when mdbgen[i] is 0 and `sdedbgalw` is 1.
-
-Debug operations in all modes are prohibited for hart i when the logical-OR of `sdedbgalw` and mdbgen[i] is 0. All halt requests from the Debug Module will remain pending, and triggers configured to enter debug mode will neither match nor fire. The register access without halting is dropped.
+==== Supervisor Domain Debug Control
+The Smsdedbg extension cite:[smmtt] introduces `sdedbgalw` field (bit 7) in CSR <<Sdseccsr,msdcfg>> to control the debuggability of supervisor domains. The `sdedbgalw` along with mdbgen[i] determines the debug allowed privilege levels, as illustrated in <<dbgpriv>>. The <<dbgaccpriv, debug access privilege>> can only be configured to debug allowed levels. 
 
 [[dbgpriv]]
 [options="header"]
-.The debuggable privilege levels per debug controls 
+[cols="25%,25%,50%"]
+.External debug allowed privilege levels per debug controls 
 |============================================
-| mdbgen[i] | sdedbgalw | Debuggable privilege levels 
-| 1      | x      | All                      
+| mdbgen[i] | sdedbgalw | Debug allowed privilege levels 
+| 1      | Don't care      | All                      
 | 0      | 1      | All except M             
-| 0      | 0      | n/a                      
+| 0      | 0      | None                      
 |============================================
 
-==== Privilege Changing Instructions
+When debug is allowed in supervisor domain, <<dbops, debug operations>> are allowed when hart executes in supervisor domain. The register access without halting the hart carries supervisor mode privilege if supported.
+
+[[dbgaccpriv]]
+==== Debug Access Privilege
+
+The *debug access privilege* is defined as the privilege level granted to the external debugger to access hardware resources with abstract commands or program buffers. Memory and register accesses from Debug Mode also carry *debug access privilege* instead of always with M-mode. The *debug access privilege* is represented by the `prv` and `v` fields in dcsr. The legal privilege levels programmable to `dcsr` in Debug Mode are elaborated in <<prvvacc>>. Debugger accesses to registers and memory will checked by permission check mechanisms against *debug access privilege*, and trigger traps if they violate corresponding rules.
```

Comment:
Previously the statement was more detailed as follows: 

> Memory and register access within Debug Mode are subject to the *debug access privilege*, with all hardware protections, including MMU, PMP, and PMA

Just to clarify, we are still claiming the same hardware protections. For example, if debug access privilege and virtual address translation enabled in that point, memory operations in Program Buffer are subject to the address translation. Am I correct?

Reply:
Yes, you are right about the address translation in program buffer. The reason we remove MMU, PMP, PMA is trying to  be more general.

Reply:
thank you for confirming it.